### ISSUE TITLE
Inquiry about using dino for new environments and for evaluation

### ISSUE THREAD

**Issue by tabz23** [gaoyuezhou/dino_wm#12]
Hello,
To use dino for a new environment, what are the steps I need to take aside from collecting a dataset in the format dino accepts, create the dataset file dino_wm/datasets and the conf file in the dino_wm/conf/env? Must i also create the environment file, such as dino_wm/env/pointmaze or dino_wm/env/pusht, training is offline, so why do we need these env files, and what else do I need to do?

Also, can you please expand on how to use the goal_source parameter in plan.py or provide an example? I deduced that it is possible to do random_state, random_action, dset and file, but I do not really understand the use case of each. it would be helpful to point me to any documentation or explain this.

Thank you very much!

**Comment by gaoyuezhou** [comment]
Hi,

If you only want to train a world model, you don't need environment files like  `dino_wm/env/pusht`. In fact, these files won't be loaded during training. They are only used during planning (`plan.py`) when you evaluate the planned actions in the real environment. 

As for the `goal_source` parameter, it specifies where the goals come from during planning.
- `goal_source='dset'`: it samples init and goal states from validation trajectories, with segment length `goal_H`. It sets the first states of the segments to be the initial states, and the last states of the segments as the goal states.
- `goal_source='random_action'`: similarly samples trajectory segments from the validation set, but replace all actions to be random actions, the actions are replayed in the environment to obtain goal states. This setting is the easiest among all modes and is not used in the paper. 
- `goal_source='random_state'`: it samples two set of random states from the environment, setting one as the initial states and the other as the goal states.
- `goal_source='file'`: it loads the init and goal states from `goal_file_path`. `plan.py` always dump the current planning targets into file `plan_targets.pkl`, which you could input to another planning run by setting `goal_source='file'` to ensure using the same initial and goal states.
